src/pages/registration/new_registration/LinkRegistration.tsx

```diff
@@ -51,65 +49,45 @@ const emptyDoiFormValues: DoiFormValues = {
   [LinkRegistrationFormFieldName.Link]: '',
 };
 
-const doiUrlBase = 'https://doi.org/';
 const doiUrlPlaceholder = `${doiUrlBase}10.1000/xyz123`;
-const doiRegExp = new RegExp('\\b(10[.][0-9]{4,}(?:[.][0-9]+)*/(?:(?!["&\'<>])\\S)+)\\b'); // https://stackoverflow.com/a/10324802
 
 export const LinkRegistration = ({ expanded, onChange }: StartRegistrationAccordionProps) => {
   const { t } = useTranslation();
-  const [doiQuery, setDoiQuery] = useState('');
   const history = useHistory();
-  const dispatch = useDispatch();
+  const [doiQuery, setDoiQuery] = useState('');
 
-  const openRegistration = async () => {
-    if (!mutateDoi.data) {
-      return;
-    }
-    history.push(getRegistrationWizardPath(mutateDoi.data.identifier), { highestValidatedTab: -1 });
+  const onCreateRegistrationSuccess = (response: AxiosResponse<Registration, any>) => {
+    history.push(getRegistrationWizardPath(response.data.identifier), { highestValidatedTab: -1 });
   };
 
-  const resultsQuery = useQuery({
-    enabled: !!doiQuery,
-    queryKey: ['doi-results', doiQuery],
-    queryFn: () => fetchResults({ doi: doiQuery }),
-  });
-
-  const searchResults = resultsQuery.data?.hits ?? [];
-
-  const mutateDoi = useMutation({
-    mutationFn: (doi: string) => getRegistrationByDoi(doi),
-    onSuccess: (response) => {
-      if (stringIncludesMathJax(response.title)) {
-        typesetMathJax();
-      }
-    },
-    onError: () => {
-      dispatch(setNotification({ message: t('feedback.error.get_doi'), variant: 'error' }));
-    },
-  });
-
-  const mutateDoiRef = useRef(mutateDoi);
+  const doiSearchResults = useGetRegistrationsWithDoi(doiQuery);
+  const doiPreview = useGetDoiPreview();
+  const createRegistrationFromDoi = useCreateRegistrationFromDoi(onCreateRegistrationSuccess);
+
+  const registrationsWithDoi = doiSearchResults.data?.hits ?? [];
+  const isLookingUpDoi = doiSearchResults.isFetching || doiPreview.isPending;
+  const isCreatingRegistration = createRegistrationFromDoi.isPending;
+
   useEffect(() => {
-    if (resultsQuery.isSuccess && resultsQuery.data.hits.length === 0) {
-      mutateDoiRef.current.mutate(doiQuery);
+    if (doiSearchResults.isSuccess && doiSearchResults.data.hits.length === 0) {
+      doiPreview.mutate(doiQuery);
     }
-  }, [resultsQuery.isSuccess, resultsQuery.data, doiQuery]);
-
-  const isLookingUpDoi = resultsQuery.isFetching || mutateDoi.isPending;
+  }, [doiSearchResults.isSuccess, doiSearchResults.data, doiQuery]);
```

Comment:
Får en warning fra ESLint her

![bilde](https://github.com/BIBSYSDEV/NVA-Frontend/assets/6313468/9fed5b13-f975-4820-aa12-8537426353c9)


```suggestion
  }, [doiSearchResults.isSuccess, doiSearchResults.data, doiQuery, doiPreview]);
```